### DISCUSSION TITLE
dietpi-backup source selection

### DISCUSSION THREAD

**Discussion by jtmoon79** [MichaIng/DietPi#5836] [Q&A] [answered]
**tl;dr** what does `dietpi-backup` backup?

Using `dietpi-backup` from 8.10.2, it's default _Filter_ has contents

```text
# Prefix "-" exclude items, "+" include items which would match a wildcard exclude rule.
# Suffix "/" match directories only, no files or symlinks
+ /mnt/dietpi_userdata/
- /mnt/*
- /media/
```
(snippet of relevant _Filter_ file lines)

The _Filter_ seems to suggest that `dietpi-backup` must be told exactly what to ignore.

Looking at directories on my DietPi system, there are paths like `/var/log` that is a `tmpfs` filesystem, the transient `/proc` System Filesystem, a `cifs` remote mount. So I'm not sure if I should add _Filter_ entries like `- /var/log`.

At the same time, I'm not sure if `dietpi-backup` _only_ attempts to backup `+` entries _or_ does it start from `/`. So I'm not sure if I should add _Filter_ entries like `+ /home`.

- Does `dietpi-backup` strictly follow _Filter_ file and _only_ backup paths marked `+`?
  Or, does `dietpi-backup` backup from `/`?
- Does `dietpi-backup` include transient and remote filesystem paths too?
- Given any path, does `dietpi-backup` restrict that path to one filesystem (`rsync --one-file-system`)?

Looking at the [docs online](https://dietpi.com/docs/dietpi_tools/#dietpi-backup-backuprestore), it's not explained.

It would an improvement if `dietpi-backup` explained it's "backup strategy" (how it chooses to backup paths and files).

**Comment by MichaIng** [comment, marked as answer]
It does backup everything `/`, excludes tmpfs locations and then applies the filters from the file. It is not restricted to one file system, otherwise `/boot` wouldn't be backed up on images where it's a dedicated filesystem.

> It would an improvement if `dietpi-backup` explained it's "backup strategy" (how it chooses to backup paths and files).

We could add the info to the `Help` menu entry and/or to the docs: https://dietpi.com/docs/dietpi_tools/#dietpi-backup-backuprestore
